client/js/analysis.js

```diff
@@ -1,25 +1,110 @@
 window.onload = function() {
-  for (let i = 1; i < 6; i++) {
+  // Listen for submission click
+  const formSubmit = document.getElementById('sendbutton');
+  formSubmit.addEventListener('click', fetchResponse, false);
+};
+
+/**
+ * Fetches the data from the backend and populates the screen
+ *
+ */
+async function fetchResponse() {
+  const param = document.getElementById('searchbar').value;
+  const response = await fetch(`/api/analysis?name=${param}`);
+  const value = await response.json();
+  console.log(value);
+  populationHandler(value);
+}
+
+/**
+ * Handles populating all elements on the screen
+ *
+ * @param {object} json the json object
+ */
+function populationHandler(json) {
+  const charts = document.getElementById('charts');
+  const list = document.getElementById('list');
+
+  removeAllChildNodes(charts);
+  removeAllChildNodes(list);
+  populateDonutCharts(json);
+  const time = populateComments(json);
+  determineOverall(json, time*1500);
+}
+
+/**
+ * Puts the charts on the screen
+ *
+ * @param {object} json the json object
+ */
+function populateDonutCharts(json) {
+  const map = json.perspectiveMap;
+  for (const key in map) {
+    if (key) { // this is here bc of the linter
+      // Convert to string for type compatibility
+      addDonutChart(key, String(100 - map[key]*100));
```

Comment:
`String` constructor should be avoided; use `.toString()` instead

Reply:
Also, why are you subtracting from 100? Are your percentages still reversed? Just use `calc()` in your CSS.

Reply:
calc messes up the keyframes